File: Algotoolbox/week2/gcd.cpp
#include <bits/stdc++.h>
using namespace std;
//Naive Algo
// int gcdSlow(int a, int b)
// {
//     int best = 0;
//     for (int i = 1; i < a + b; i++)
//     {
//         if (a % i == 0 && b % i == 0)
//         {
//             best = i;
//         }
//     }
//     return best;
// }
int gcdFast(int a, int b)
{
    if (b == 0)
    {
        return a;
    }

    int A = a % b;
    return gcdFast(b, A);
}
int main()
{
    int a,b; cin >> a >> b;
    cout << gcdFast(a, b);

    return 0;
}

File: Algotoolbox/week2/gcd.py
# Naive Algorithm
def gcdSlow(a,b):
    best =0
    for i in range(1,a+b):
        if(a%i==0 and b%i==0):
            best = i
    return best

def gcdFast(a,b):
    if b ==0:
        return a
    
    A = a%b
    return gcdFast(b,A)

result = gcdSlow(10,4)
res2 = gcdFast(10,4)
print(f"First Ans is {result} and second is {res2}")